### ISSUE TITLE
Debugging language server extension doesn't work

### ISSUE THREAD

**Issue by MastroLindus** [Microsoft/vscode#16365]
- VSCode Version:  1.6.1
- OS Version: Linux Mint 17
- Node 7 (same with 4 and 6)

1)Create an extension using the client/server model as described in the documentation
2)Try to debug the server as describer in the documentation https://code.visualstudio.com/Docs/extensions/example-language-server by pressing f5/attaching node
3)Debugging doesn't work. The server goes to debug mode (you see the top bar with the pause/disconnect/etc buttons), you can set breakpoints, but they never get triggered. Debug console doesn't show anything.

Debugging the client and/or debugging a normal extension without the client/server model works as expected.

My codebase is very similar to the example codebase found in the docs, so it should be easy to reproduce. The server is working (I can change for example an autocompletion entry, reload the extension host environment and see my new entry there) but I have no way to either debug or output anything so developing is very painful.

Not sure whether it's a real bug or simply the docs are outdated and something else is missing to make this work.

**Comment by siegebell** [comment]
dup of #16143?

**Comment by chrmarti** [comment]
@MastroLindus Could you append your launch.json?

**Comment by MastroLindus** [comment]
@chrmarti  i can reproduce with the standard example repository https://github.com/Microsoft/vscode-languageserver-node-example?files=1

PS I don't think it's a duplicate of #16143 since that one seems windows-related

**Comment by weinand** [comment]
@MastroLindus I could not reproduce this problem on linux (elementary OS 0.4 Loki, Built on "Ubuntu 16.04 LTS"), node v6.9.1, VS Code 1.6.1 and 1.7.2.

My test scenario was:
- opening VS Code on "client" folder and debugging extension with "F5"
- in the new Extension Host window I opened an empty folder and created a new file foo.txt and entered some text including the word "typescript". This immediately got a squiggly (which indicates that the server is running).
- opened "server" folder and set breakpoint in line 45 of server.ts (onDidChangeContent).
- editing foo.txt made the debugger hit the breakpoint.

Please try the scenario from above to make sure that we are doing the same steps.

**Comment by apazureck** [comment]
I have the same problem here. I cannot debug my language server I created deriving from the server on github mentioned in the documentation (see @MastroLindus post).

After that I tried the original project and the same here -> no debugging on the language server is possible, no breakpoints are hit.

VSCode 1.7.2, Windows 10, Node 6.9.1

also tried to play around with debug ports. did nothing for me :(
